### DISCUSSION TITLE
Verbindungsabbrüche bei Limitierung mehrerer Wechselrichter

### DISCUSSION THREAD

**Discussion by DerBaer1986** [tbnobody/OpenDTU#1191] [Q&A]
Hi,

Ich hab jetzt lange rum getan und viele Sachen versucht. Leider alles ohne Ergebnis.
Ich hab 3 HM-300 die ich über Node Red limitieren möchte. Grundsätzlich geht das auch, aber nach ein paar Sekunden kommunizieren die Wechselrichter nicht mehr und es wird nix mehr geändert.

Der Aufbau sieht so aus, dass ich via Home Assistent einen Shelly auslese und jeden einzelnen Wert mit Node Red an den Open DTU via MQTT sende. Grundsätzlich hatte ich mal alle 10 Sek eingestellt.

Nun werden für alle 3 Wechselrichter ca die ersten 2 Limits gesetzt und danach ist die Kommunikation im Nirvana. Nur ein Neustart des DTU hilft dann und auch nur wenn ich die Nodes deaktiviere.

Ich hab versucht alle 3 Wechselrichter an einem DTU laufen zu lassen und an 3 DTUs. Auch wenn ich 3 unterschiedliche DTUs verwende gibt's Probleme. Irgendwann verschwindet einer und lässt sich nicht mehr limitieren oder auslesen.

Ich hab auch schon mit verschiedensten Intervallen rumgespielt, so dass jeder Wechselrichter zb einen eigenen Intervall bekommt. Zb 1. Alle 7, 2. Alle 10 und 3 alle 13.

Inzwischen bin ich ziemlich frustriert wie ihr euch sicher vorstellen könnt.

![Screenshot_20230801-141848.png](https://github.com/tbnobody/OpenDTU/assets/24823860/e6ae9c7c-0b4d-469f-b2cc-7cb428e77ca1)

Um auf die Hardware einzugehen:

-  Platine hier aufm Git
-  AZDelivery ESP32 Dev Kit C V4 NodeMCU
-  DollaTek NRF24L01 + PA + LNA Funk-Transceiver-Modul 2.4G 1100m mit Antenne
-  0.96 Zoll Display
-  100uF Elko

Open DTU Version

![Screenshot_20230801-165455.png](https://github.com/tbnobody/OpenDTU/assets/24823860/87aa90c2-8773-4dc7-a83e-aa80c62f9f05)

**Comment by Omega13x** [comment]
Das liegt sehr wahrscheinlich daran, das wenn ein WR den Bedarf deckt, die anderen das Limit "0" bekommen. Das heißt für die anderen Wechselrichter, abschalten. Es dauert dann zw. 4 und 10 Minuten bis der/die Wechselrichter sich wieder einschalten. Du musst verhindern das sich die Wechselrichter ausschalten, also einen Mindestwert festlegen, auch wenn dann etwas Strom ins Netz verschenkt wird.

**Reply by DerBaer1986** [reply]
Daran hab ich noch nicht gedacht. Ich war zwar der Meinung das ich eh einen gefilterten Kanal verwende. Ich hab jetzt grad nachgesehen und festgestellt das dies nicht der Fall ist.
Danke für den Hinweis!

**Reply by DerBaer1986** [reply]
OK leider kann ich sagen daran lag jetzt ned.
Ich hab die Wechselrichter eingescha und ne Minute gewartet bis die Limitierung läuft. Beide wurden von 5% auf 100% gesetzt da so viel geraucht wurde. Letzten Endes das gleiche Bild wie vorher, mindestens einer der WR ist nicht mehr erreichbar.
Die Regelung sendet natürlich immer noch alle 10 Sek die 100%.

**Reply by Omega13x** [reply]
Ich habe hier auch zwei WR. Einen HM-600 und einen HM-300. Ich hatte mit denen noch nie Probleme. Das ganze läuft bei mir schon seit Anfang April. Geregelt wird mit Node Red in Home Assistant.
Hast du vielleicht manchmal keine Werte vom Shelly. So das "NaN" angezeigt wird?

**Reply by DerBaer1986** [reply]
Websocket: [/livedata][3] disconnect
16:00:10.938 > RX Period End
16:00:10.938 > Success
16:00:11.003 > TX ActivePowerControl Channel: 23 --> 51 84 70 59 88 80 10 94 84 81 0B 00 0B B8 00 00 61 83 2F 
16:00:12.929 > RX Period End
16:00:12.930 > All missing
16:00:12.930 > Nothing received, resend whole request
16:00:12.930 > TX ActivePowerControl Channel: 40 --> 51 84 70 59 88 80 10 94 84 81 0B 00 0B B8 00 00 61 83 2F 
16:00:13.112 > Interrupt received
16:00:13.296 > RX Channel: 61 --> D1 84 70 59 88 84 70 59 88 81 00 00 0B 00 14 07 48 | -80 dBm
16:00:14.937 > RX Period End
16:00:14.937 > Success
16:00:15.244 > TX ActivePowerControl Channel: 61 --> 51 84 71 32 73 80 10 94 84 81 0B 00 0B B8 00 00 61 83 BE 
16:00:15.354 > Received MQTT message on topic: nachtspeisung/112184713273/cmd/limit_nonpersistent_absolute
16:00:15.354 > Limit Non-Persistent: 300 W
16:00:15.422 > Received MQTT message on topic: nachtspeisung/112184705988/cmd/limit_nonpersistent_absolute
16:00:15.422 > Limit Non-Persistent: 300 W
16:00:16.974 > RX Period End
16:00:16.974 > All missing
16:00:16.974 > Nothing received, resend whole request
16:00:16.974 > TX ActivePowerControl Channel: 75 --> 51 84 71 32 73 80 10 94 84 81 0B 00 0B B8 00 00 61 83 BE 
16:00:17.110 > Interrupt received
16:00:17.286 > RX Channel: 23 --> D1 84 71 32 73 84 71 32 73 81 00 00 0B 00 14 07 48 | -80 dBm
16:00:19.177 > RX Period End
16:00:19.177 > Success
16:00:19.239 > TX ActivePowerControl Channel: 3 --> 51 84 70 59 88 80 10 94 84 81 0B 00 0B B8 00 00 61 83 2F 
16:00:21.204 > RX Period End
16:00:21.204 > All missing
16:00:21.204 > Nothing received, resend whole request
16:00:21.204 > TX ActivePowerControl Channel: 23 --> 51 84 70 59 88 80 10 94 84 81 0B 00 0B B8 00 00 61 83 2F

**Reply by DerBaer1986** [reply]
![Screenshot_20230801-160132.png](https://github.com/tbnobody/OpenDTU/assets/24823860/33e197ae-a441-4ee4-bd30-d9a70f0f9222)

**Reply by DerBaer1986** [reply]
![Screenshot_20230801-160255.png](https://github.com/tbnobody/OpenDTU/assets/24823860/1a674824-c9a3-4303-911b-6bf907087160)

**Reply by DerBaer1986** [reply]
Ne inzwischen nutze ich einen gefilterten Kanal, aber auch davor waren immer passende Werte da.

Es lief vorher auch schon mal via Python aber da war das Problem das selbe und ich hatte gehofft mit NodeRed geht's besser.

**Reply by Omega13x** [reply]
Wie hast du die Mqtt Zeit und wie die DTU Zeit eingestellt? Du musst vermeiden das diese kollidieren. Also Mqtt z.B. auf 5Sek und DTU auf 7Sek. So kollidieren sie nur bei jeder 5ten Übertragung. Damit hatte ich auch mal zu kämpfen. Wichtig ist auch ein ausreichender Abstand der DTU zu anderen WLan-Geräte. Besonders zum Router.

**Reply by DerBaer1986** [reply]
DTU is auf 3 und MQTT auf 5.

WLAN Geräte sind sonst keine da außer die Batterie.

Aber grundsätzlich funktioniert die Kommunikation ja. Erst wenn ich die Limitierung über NodeRed einschalte gehen die Probleme los.

Ich weiß auch sicher das NodeRed nur einen Wert alle 10 Sek pro WR ausgiebt. Dafür aber gleichzeitig auf alle 3. Ich hatte aber auch da schon mit unterschiedlichen Zeiten rumgespielt und das hat nix gebracht.

**Reply by DerBaer1986** [reply]
![Screenshot_20230801-163134.png](https://github.com/tbnobody/OpenDTU/assets/24823860/94b20107-c2c8-4160-9e34-97c17794eb7c)

**Comment by DPO99** [comment]
Vermutlich ist das trotzdem einfach zu viel. Ich habe auch 2 Wechselrichter in der Steuerung (eigentlich nicht parallel aber manchmal kommen sie sich eben doch in die quere) und da gibts dann auch gerne mal nen fehler. Installiere doch einfach 3 DTU's. Das kost doch nicht die Welt und sollte ggf. das Problem lösen.

**Reply by DerBaer1986** [reply]
Was genau meinst du mit zuviel?

Ich hatte ja eben auch schon drei DTUs am laufen und das funktioniert genauso wenig.

**Reply by DPO99** [reply]
Naja wenn alle 10 Sekunden Befehle für 3 Wechselrichter an die DTU gehen, dann wird das zu viel sein um es innerhalb von der Zeit abzuarbeiten und direkt wieder die nächsten Befehle anzunehmen. Ich bin auch auf 30 Sekunden für einen WR hoch und selbst da kann es mal passieren, dass der WR einen Befehl nicht abarbeitet. Erhöhe doch einfach mal die Zeit deutlich und nimm nur einen WR. Dann kann man sich an das mögliche/funktionierende rantasten.

**Reply by DerBaer1986** [reply]
Naja wie schon beschrieben, eine WR alleine funktioniert einwandfrei, sogar dauerhaft mit 5 Sek Intervallen. Sobald aber ein 2. dazu kommt geht nix mehr. Auch mit 30 Sek Intervallen ned.

**Comment by kjmwwi** [comment]
Ich hatte das selbeProblem. 3 WR über Node Red und keine Antwort . Bei mir stieg in OpenDTU (ohne Node Red) die Zykluszeit bis alle 3 WR abgefragt waren auf 15 sek an. 5 sek pro WR.
Ich habe die Node Red Zeit auf etwas mehr als den Faktor 2 gestellt => 33sek. Nun läuft es.

Ich weiß nicht ob man die Zykluszeit von OpenDTU beschleunigen kann. Dann müsste es schneller gehen.

**Reply by DerBaer1986** [reply]
Ja ich hab leider ähnliches erlebt. Aktuell läufts bei mir bei 25 Sek stabil. Meine Zykluszeiten sind eh schon sehr niedrig, aber der DTU bzw der WR aktualisiert nicht schneller, auch wenn ich die Zykluszeit reduziere.
